internal/provider/providers/goip/provider.go

```diff
@@ -0,0 +1,170 @@
+		return fmt.Errorf("%w", errors.ErrPasswordNotSet)
+	}
+	return nil
+}
+
+func (p *Provider) String() string {
+	return utils.ToString(p.domain, "@", constants.GoIP, p.ipVersion)
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Host() string {
+	return "@"
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) IPv6Suffix() netip.Prefix {
+	return p.ipv6Suffix
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return p.domain
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Host:      p.Host(),
+		Provider:  "<a href=\"https://www.goip.de/\">GoIP.de</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
+	u := url.URL{
+		Scheme: "https",
+		User:   url.UserPassword(p.username, p.password),
+		Host:   "www.goip.de",
+		Path:   "/setip",
+	}
+	values := url.Values{}
+	values.Set("subdomain", p.domain)
+	values.Set("username", p.username)
+	values.Set("password", p.password)
+	values.Set("shortResponse", "true")
+	useProviderIP := p.useProviderIP && (ip.Is4() || !p.ipv6Suffix.IsValid())
+	if !useProviderIP {
+		values.Set("ip", ip.String())
+	}
+	u.RawQuery = values.Encode()
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("creating http request: %w", err)
+	}
+	headers.SetUserAgent(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("doing http request: %w", err)
+	}
+	defer response.Body.Close()
+
+	switch response.StatusCode {
+	case http.StatusOK:
+	case http.StatusNoContent:
+		return ip, nil
+	case http.StatusUnauthorized:
+		return netip.Addr{}, fmt.Errorf("%w", errors.ErrAuth)
+	case http.StatusConflict:
+		return netip.Addr{}, fmt.Errorf("%w", errors.ErrZoneNotFound)
+	case http.StatusGone:
+		return netip.Addr{}, fmt.Errorf("%w", errors.ErrAccountInactive)
+	case http.StatusLengthRequired:
+		return netip.Addr{}, fmt.Errorf("%w: %s", errors.ErrIPSentMalformed, ip)
+	case http.StatusPreconditionFailed:
+		return netip.Addr{}, fmt.Errorf("%w: %s", errors.ErrPrivateIPSent, ip)
+	case http.StatusServiceUnavailable:
+		return netip.Addr{}, fmt.Errorf("%w", errors.ErrDNSServerSide)
+	default:
+		return netip.Addr{}, fmt.Errorf("%w: %d: %s",
+			errors.ErrHTTPStatusNotValid, response.StatusCode, utils.BodyToSingleLine(response.Body))
+	}
+
+	b, err := io.ReadAll(response.Body)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("reading response body: %w", err)
+	}
+	s := string(b)
+	switch {
+	case strings.HasPrefix(s, p.domain+" ("+ip.String()+")"):
+		return ip, nil
+	case strings.HasPrefix(s, "Zugriff verweigert"):
```

Comment:
to avoid caring about uppercase/lowercase, in case they change that too

```suggestion
	case strings.HasPrefix(strings.ToLower(s), "zugriff verweigert"):
```

Reply:
Implemented.